spec/requests/audio_recordings_spec.rb

```diff
@@ -142,4 +142,180 @@
+      ])
+    end
+
+    it 'can filter for recordings by exact start time only (6 PM local time)' do
+      post_filter(filter: {
+        eq: {
+          recorded_date: ['local_offset', 'tod', '18:00']
+        }
+      })
+
+      # we should find recordings from
+      # +10:00 Sydney & +08:00 Perth
+      # and after DST, +11:00 Sydney & +08:00 Perth (no DST in Perth)
+      # but since local_offset is used we get consistent selections back
+      expect(api_data).to match([
+        # these are all +10:00 times
+        make_result(site_id: sydney.id, day: 1, hour: 18),
+        make_result(site_id: perth.id, day: 1, hour: 16),
+        make_result(site_id: sydney.id, day: 2, hour: 18),
+        make_result(site_id: perth.id, day: 2, hour: 16),
+        make_result(site_id: sydney.id, day: 3, hour: 18),
+        make_result(site_id: perth.id, day: 3, hour: 16),
+
+        # DST cross-over
+        make_result(site_id: sydney.id, day: 4, hour: 18),
+        make_result(site_id: perth.id, day: 4, hour: 16),
+        make_result(site_id: sydney.id, day: 5, hour: 18),
+        make_result(site_id: perth.id, day: 5, hour: 16),
+        make_result(site_id: sydney.id, day: 6, hour: 18),
+        make_result(site_id: perth.id, day: 6, hour: 16)
+      ])
+    end
+
+    it 'can filter for recordings by exact start time only (6 PM local time), adjusting for DST' do
+      post_filter(filter: {
+        eq: {
+          recorded_date: ['local_tz', 'tod', '18:00']
+        }
+      })
+
+      # we should find recordings from
+      # +10:00 Sydney & +08:00 Perth
+      # and after DST, +11:00 Sydney & +08:00 Perth (no DST in Perth)
+      # however, since local_tz is used we get inconsistent selections back
+      expect(api_data).to match([
+        # these are all +10:00 times
+        make_result(site_id: sydney.id, day: 1, hour: 18),
+        make_result(site_id: perth.id, day: 1, hour: 16),
+        make_result(site_id: sydney.id, day: 2, hour: 18),
+        make_result(site_id: perth.id, day: 2, hour: 16),
+        make_result(site_id: sydney.id, day: 3, hour: 18),
+        make_result(site_id: perth.id, day: 3, hour: 16),
+
+        # DST cross-over - now no recordings in Sydney start at 18:00
+        make_result(site_id: perth.id, day: 4, hour: 16),
+        make_result(site_id: perth.id, day: 5, hour: 16),
+        make_result(site_id: perth.id, day: 6, hour: 16)
+      ])
+    end
+
+    def post_filter(filter:)
+      body = {
+        filter:,
+        projection: { include: [:recorded_date_timezone, :recorded_date, :site_id] },
+        sorting: { orderBy: 'recorded_date', direction: 'asc' }
+      }
+
+      post '/audio_recordings/filter', params: body, **api_with_body_headers(reader_token)
+
+      expect_success
+    end
+
+    def make_result(site_id:, day:, hour:)
+      a_hash_including({
+        site_id:,
+        recorded_date: Time.new(2021, 10, day, hour, 0, 0, '+10:00'),
```

Comment:
Does this get converted back to UTC? I am under the impression (although i may be wrong) that recorded_date should be in UTC time, and `recorded_date_timezone` is used to convert

Reply:
I haven't run the tests yet, but all the dates will be compared to by their instant (as in no matter what offset they're in two things that happened in the real world at the same time will be  equivalent).

For the tests I chose to work in Brisbane time over UTC just because it is easier to reason a timezone I'm more familiar with. But the tests would work exactly the same way with UTC, just the numbers would be, IMO, harder to understand.